modules/nlu-testing/src/views/full/TestModal.tsx

```diff
@@ -15,38 +15,81 @@ interface Props {
   onTestCreated: (test: any) => void
 }
 
-const noneIntent = {
+const TEST_ALL_CTX = '*'
+const NONE_CTX = 'none'
+const NONE_INTENT = {
   name: 'none',
   slots: []
 } as sdk.NLU.IntentDefinition
 
-export const CreateTestModal: FC<Props> = props => {
+export const TestModal: FC<Props> = props => {
   const [utterance, setUtterance] = useState<string>((props.test && props.test.utterance) || '')
-  const [expectedIntent, setTargetIntent] = useState<sdk.NLU.IntentDefinition>(noneIntent)
+  const [expectedIntent, setExpectedIntent] = useState<sdk.NLU.IntentDefinition>(NONE_INTENT)
   const [intents, setIntents] = useState<sdk.NLU.IntentDefinition[]>([])
   const [availableCtx, setAvailableCtxs] = useState([])
   const [testingCtx, setTestingCtx] = useState('*')
   const [slotConditions, setSlotConditions] = useState<_.Dictionary<string>>({})
   const [expectedCtx, setExpectedCtx] = useState<string>('')
+  const [isEditing, setIsEditing] = useState(false)
 
   useEffect(() => {
     // tslint:disable-next-line: no-floating-promises
     props.api.fetchIntents().then(intents => {
-      setIntents([...intents, noneIntent])
+      setIntents([...intents, NONE_INTENT])
       const ctxs = _.chain(intents)
         .flatMap(i => i.contexts)
         .uniq()
         .value()
-      setAvailableCtxs([...ctxs, '*', 'none'])
+      setAvailableCtxs([...ctxs, TEST_ALL_CTX, NONE_CTX])
       setExpectedCtx(ctxs[0])
     })
   }, [])
 
+  useEffect(() => {
+    const { test } = props
```

Comment:
```suggestion
    const { test } = props
    
```

just a preference, but I like separating code blocks with an empty line (variable definition, if blocks, etc.)